packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -4655,4 +4723,127 @@ describe('TransactionController', () => {
+  describe('initTrackingMap', () => {
+    // eslint-disable-next-line jest/no-done-callback
+    it('should initialize the tracking map on construction', (done) => {
+      const hub = new EventEmitter() as TransactionControllerEventEmitter;
+      hub.on('tracking-map-init', () => {
+        done();
+      });
+      const controller = newController({
+        options: {
+          hub,
+        },
+      });
+      expect(controller).toBeDefined();
+    });
+    it('should handle removals in the networkController registry', (done) => {
+      const hub = new EventEmitter() as TransactionControllerEventEmitter;
+      let stateChangeFn: (state: any) => void;
+      const mockGetNetworkClientRegistry = jest.fn();
+      mockGetNetworkClientRegistry.mockImplementationOnce(() => ({
+        sepolia: {
+          configuration: {
+            chainId: SEPOLIA.chainId,
+          },
+        },
+        goerli: {
+          configuration: {
+            chainId: GOERLI.chainId,
+          },
+        },
+        'customNetworkClientId-1': {
+          configuration: {
+            chainId: '0xa',
+          },
+        },
+      }));
+      hub.on('tracking-map-init', () => {
+        mockGetNetworkClientRegistry.mockClear();
+        mockGetNetworkClientRegistry.mockImplementationOnce(() => ({
+          sepolia: {
+            configuration: {
+              chainId: SEPOLIA.chainId,
+            },
+          },
+          goerli: {
+            configuration: {
+              chainId: GOERLI.chainId,
+            },
+          },
+        }));
+      });
+      hub.on('tracking-map-remove', () => {
+        done();
+      });
+      const controller = newController({
+        options: {
+          getNetworkClientRegistry: mockGetNetworkClientRegistry,
+          onNetworkStateChange: (fn: (state: any) => void) => {
+            stateChangeFn = fn;
+          },
+          hub,
+        },
+      });
+      expect(controller).toBeDefined();
+    });
+  });
+  // eslint-disable-next-line jest/no-done-callback
+  it('should handle removals in the networkController registry', (done) => {
+    const hub = new EventEmitter() as TransactionControllerEventEmitter;
+    let stateChangeFn: (state: any) => void;
+    const mockGetNetworkClientRegistry = jest.fn();
+    mockGetNetworkClientRegistry.mockImplementationOnce(() => ({
+      sepolia: {
+        configuration: {
+          chainId: SEPOLIA.chainId,
+        },
+      },
+    }));
+    hub.on('tracking-map-init', () => {
+      mockGetNetworkClientRegistry.mockClear();
+      mockGetNetworkClientRegistry.mockImplementationOnce(() => ({
+        sepolia: {
+          configuration: {
+            chainId: SEPOLIA.chainId,
+          },
+        },
+        goerli: {
+          configuration: {
+            chainId: GOERLI.chainId,
+          },
+        },
+      }));
+    });
+    hub.on('tracking-map-add', ([trackedNetworkClientId]) => {
+      expect(trackedNetworkClientId).toBe('goerli');
```

Comment:
i think we should assert over the array as a whole rather than the first item

Reply:
done here https://github.com/MetaMask/core/pull/3770/commits/abdb8965acbd21d8e131cdc2cc1fb7e50db6d469